docs/src/wallet_to_wallet_with_tariscript.md

```diff
@@ -0,0 +1,205 @@
+
+For a full breakdown of TariScript please see this [RFC](https://rfc.tari.com/RFC-0201_TariScript.html). This document will 
+repeat some of the detail contained in the RFC so that it is self-contained.
+
+We will use the same [notation](https://rfc.tari.com/RFC-0201_TariScript.html#notation) described in the RFC.
+
+## Alice sends a transaction to Bob
+Alice is going to send a transaction to Bob which spends a commitment \\( C_a \\), with a spending script \\( \alpha_a \\), 
+to a commitment \\( C_b \\), with a spending script \\( \alpha_b \\) that Bob will use to spend the new UTXO in the future.
+
+### Alice Round 1
+To spend \\( C_a \\) Alice must know the following:
+1. Spending key \\( k_a \\) of \\( C_a \\),
+2. Value \\( v_a \\) of \\( C_a \\),
+3. Features  \\( F_a \\) of \\( C_a \\),
+4. Script \\( \alpha_a \\) whose hash \\( \sigma_a = H(\alpha_a) \\),
+5. Script input \\( \theta_a \\) for script \\( \alpha_a \\),
+6. Script key \\( k_{sa} \\) whose public value (\\( K_{sa} = k_{sa} \cdot G \\)) is the key left over when script \\( \alpha_a \\) is executed with input \\( \theta_a \\)   
+7. Height \\( h_a \\) where \\( C_a \\) was mined.
+
+Alice then chooses the following values:
+1. Amount \\( v_b \\) being sent to Bob
+2. Fee value \\( f \\). The amounts need to balance out so \\( v_a = v_b + f \\)    
+3. A random nonce \\( r_a \\) whose public value is \\( R_a = r_a \cdot G \\)
+4. An UTXO offset key for Bob \\( k_{Ob} \\) whose public value is \\( K_{Ob} = k_{Ob} \cdot G \\)
+5. The script \\( \alpha_b \\) that will become the spending script on \\( C_b \\).
+6. Change spending key \\( k_c \\) for the change output \\( C_c \\)
+8. A script for the change UTXO \\( \alpha_c \\)
+9. A script key \\( k_{sc} \\) for \\( \alpha_c \\)    
+10. The UTXO offset key for the change UTXO \\( k_{Oc} \\) whose public value is \\( K_{Oc} = k_{Oc}G \\)
+ 
+Firstly Alice prepares the transaction the same way she would for a vanilla MW transaction by calculating the excess:
+\\( x_s = k_c - k_a \\)
+
+Alice will also calculate the rangeproof \\( RP_c \\) for her change output for the modified commitment
+
+\\( \hat{C}_c = (k_c + \beta_c) \cdot G + v_c \cdot H \\)
+
+where
+
+\\( \beta_c = H(\sigma_c || F_c || K_{Oc}) \\).
+
+Alice then sends the following values to Bob:
+
+| Symbol  | Description                                      |
+|---------|--------------------------------------------------                                   |
+| \\( R_a \\)       | Alice's Public Nonce                                                      |
+| \\( v_b \\)       | Amount being sent to Bob                                                  |
+| \\( f \\)         | fee                                                                       |
+| \\( m \\)         | Transaction metadata (currently just lockheight)                          |
+| \\( \sigma_b \\)  | Hash of \\( C_b \\)'s spending script,  \\( \sigma_b = H(\alpha_b) \\)    |
```

Comment:
Does this not need to be the full script. Although only the hash is required at this time, to spend it in the future, he will need to know the script.

Reply:
Ja, this was taken direct from the RFC example where the script is assumed NO_OP. I agree with this and the comments below address it further.